### PR TITLE
Masking the social security number component

### PR DESCRIPTION
**Pull request by k80bowman**
## Description
[Feature Request #361](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/361)

Adding masking of the first five digits of Social Security Numbers.

## Testing done
Added unit tests.
Tested locally using this form: http://localhost:3001/ask-a-question/veteran-information

## Screenshots
Masked SSN
image

Masked partial SSN
image

Masking and unmasking SSN with focus
https://user-images.githubusercontent.com/12739849/111170781-b25ee580-857a-11eb-8571-51c8bfe79aac.mov

## Acceptance criteria
- [ ] User is able to enter SSN as all one string, with dashes, or with spaces
- [ ] When the user is entering or editing the SSN field, the SSN appears as they entered it (one string, with dashes, with spaces)
- [ ] When the user navigates away from the SSN field (onBlur), the SSN appears with dashes and the first five digits masked
- [ ] If the user has entered fewer than six digits and then navigates away from the SSN field (onBlur), all digits appear masked
- [ ] If the user has entered between 6 and 8 digits and then navigates away from the SSN field (onBlur), the first five digits are masked and any extra digits on the end of the SSN are shown.
- [ ] Unit tests have been added that cover the above scenarios

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/src/js/widgets/SSNWidget.jsx [hunk 1]
```diff
@@ -1,33 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
 import TextWidget from './TextWidget';
 
+import { formatSSN } from 'platform/utilities/ui';
+
+/**
+ * Mask a SSN, but leave the final sequence of digits visible (up to 4)
+ */
+function maskSSN(ssnString = '') {
```
Comment (timwright12) [inline src/platform/forms-system/src/js/widgets/SSNWidget.jsx:9]: pinging @1Copenut for eyes here

**Comment by k80bowman** [comment]
Found an issue with the error message that appears onBlur. Fixed. Here is an updated screenshot of what it should look like when the user navigates away from the SSN field after entering a partial SSN:
image

File: src/platform/forms-system/src/js/widgets/SSNWidget.jsx [hunk 2]
```diff
@@ -1,33 +1,56 @@
...
-        {...this.props}
-        value={this.state.val}
-        onChange={this.handleChange}
-      />
-    );
-  }
+
+  const handleBlur = () => {
+    setDisplayVal(maskSSN(val));
+    props.onBlur(props.id);
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/widgets/SSNWidget.jsx:40]: :detective:

**Review by bkjohnson — APPROVED** [review]
Good test coverage!

**Review by 1Copenut — APPROVED** [review]
LGTM! I tested briefly on localhost with VoiceOver, JAWS (including IE11) and NVDA using the non-logged-in healthcare form. The component behaved as expected for all combinations.

**Review by timwright12 — APPROVED** [review]

**Pull request merged**